src/mlmc/sampler.py

```diff
@@ -0,0 +1,181 @@
+import os
+import time
+import numpy as np
+from typing import List
+
+from sample_storage import SampleStorage
+from sampling_pool import SamplingPool
+from new_simulation import Simulation
+from workspace import WithoutWorkspace, SimulationWorkspace, WholeWorkspace, PBSWorkspace
+
+
+class Sampler:
+
+    def __init__(self, sample_storage: SampleStorage, sampling_pool: SamplingPool, sim_factory: Simulation,
+                 n_levels: int, step_range: List[int], work_dir=None):
+        """
+
+        :param sample_storage: store scheduled samples, results and result structure
+        :param sampling_pool: calculate samples
+        :param sim_factory: generate samples
+        :param n_levels: number of MLMC levels
+        :param step_range: simulation step range
+        :param work_dir: working directory
+        """
+        self._sample_storage = sample_storage
+        self._sampling_pool = sampling_pool
+
+        self._n_levels = n_levels
+        self._sim_factory = sim_factory
+        self._step_range = step_range
+
+        # Number of created samples
+        self._n_created_samples = np.zeros(self._n_levels)
+
+        # Number of target samples
+        self.n_target_samples = np.zeros(self._n_levels)
+        # Number of planned samples
+        self.n_planned_samples = np.zeros(self._n_levels)
+        # Number of finished and collected samples
+        self.n_collected_samples = np.zeros(self._n_levels)
```

Comment:
This should rather be retrieved from the storage, so just single place to store the information.